hack/prune-libraries.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Identifies go_library() bazel targets in the workspace with no dependencies.
+# Deletes the target and its files. Cleans up any :all-srcs references
+
+# By default just checks for extra libraries, run with --fix to make changes.
+
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Output every go_library rule in the repo
+find-go-libraries() {
+  # Excluding generated //foo:go_default_library~library~
+  bazel query \
+    'kind("go_library rule", //...)' \
+    except 'attr("name", ".*~library~", //...)'
+}
+
+# Output the target if it has only one dependency (itself)
+is-unused() {
+  local n="$(bazel query "rdeps(//..., \"$1\", 1)" | wc -l)"
```

Comment:
Don't mix scoping and evaluation or scoping will overwrite the return code of evaluation:

```sh
local n
n=$(...)
```

Reply:
Done